automated/linux/ssuite/run-bench.sh

```diff
@@ -0,0 +1,163 @@
+export RESULT_FILE
+
+TESTS="throughput replayed-startup"
+TEST_DEV=sda
+FORMAT=no
+S_VERSION=
+S_URL=https://github.com/Algodev-github/S
+S_PATH="$(pwd)/S"
+
+usage() {
+	echo "\
+	Usage: [sudo] ./run-bench.sh [-t <TESTS>] [-d <TEST_DEV>] [-f <FORMAT>]
+				     [-v <S_VERSION>] [-u <S_URL>] [-p <S_PATH>]
+
+	<TESTS>:
+	Set of tests: 'throughput' benchmarks throughput, while
+	'replayed-startup' benchmarks the start-up times of popular
+	applications, by replaying their I/O. The replaying saves us
+	from meeting all non-trivial dependencies of these applications
+	(such as having an X session running). Results are
+	indistinguishable w.r.t. to actually starting these applications.
+	Default value: \"throughput replayed-startup\"
+
+	<TEST_DEV>:
+	Target device/partition: device/partition on which to
+	execute the benchmarks. If a partition is specified, then
+	the partition must contain a mounted filesystem. If a device
+	(actual drive) is specified, then that drive must contain a
+	partition ${TEST_DEV}1 in it, with a mounted fs in that
+	partition. In both cases, test files are created in that
+	filesystem.
+	Default value: sda
+
+	<FORMAT>:
+	If this parameter is set to yes and TEST_DEV points to an actual
+	drive, but the drive does not contain a mounted partition, then
+	the drive is formatted, a partition with an ext4 fs is created on
+	the drive, and that fs is used for the test.
+	Default value: no
+
+	<S_VERSION>:
+	If this parameter is set, then the S suite is cloned. In
+	particular, the version of the suite is set to the commit
+	pointed to by the parameter. A simple choice for the value of
+	the parameter is, e.g., HEAD. If, instead, the parameter is
+	not set, then the suite present in S_PATH is used.
+
+	<S_URL>:
+	If this parameter is set, then the S suite is cloned
+	from the URL in S_URL. Otherwise it is cloned from the
+	standard repository for the suite. Note that cloning is done
+	only if S_VERSION is not empty
+
+	<S_PATH>:
+	If this parameter is set, then the S suite is cloned or
+	looked for in S_PATH. Otherwise it is cloned to $(pwd)/S"
+
+	exit 1
+}
+
+while getopts "ht:d:f:p:u:v:" opt; do
+	case $opt in
+		t)
+			TESTS="$OPTARG"
+			;;
+		d)
+			TEST_DEV="$OPTARG"
+			;;
+		f)
+			FORMAT="$OPTARG"
+			;;
+		v)
+			S_VERSION="$OPTARG"
+			;;
+		u)
+			S_URL="$OPTARG"
+			;;
+		p)
+			S_PATH="$OPTARG"
+			;;
+		h)
+			usage
+			exit 0
+			;;
+	esac
+done
+install() {
+	dist_name
+	case "${dist}" in
+		debian|ubuntu)
+			pkgs="fio sysstat libaio-dev gawk coreutils bc \
+				  psmisc g++ git"
+			install_deps "${pkgs}" "${SKIP_INSTALL}"
+			;;
+		fedora|centos)
+			pkgs="fio sysstat libaio-devel gawk coreutils bc \
+				  psmisc gcc-c++ git-core"
+			install_deps "${pkgs}" "${SKIP_INSTALL}"
+			;;
+		# When build do not have package manager
+		# Assume dependencies pre-installed
+		*)
+			;;
+	esac
+}
+
+run_test() {
+	if [[ "$S_VERSION" != "" && ( ! -d $S_PATH || -d $S_PATH/.git ) ]]; then
```

Comment:
As the S_VERSION var is empty by default,  ./run-bench.sh wouldn't do anything exception the above pkgs installation. Is this desired? 

Maybe set S_VERSION to master by default? 

IMO, instead of relay on S_VERSION, we should use 'SKIP_INSTALL' to control the whole installation including installing pkgs and git clone.